dashboard/app/models/contact_rollups_pardot_memory.rb

```diff
@@ -66,9 +66,19 @@ def self.create_new_pardot_prospects
     save_sync_results(submissions, errors, Time.now) if submissions.present?
   end
 
-  # TODO: sync contacts that change pardot mappings
-  # TODO: sync contacts with updated content
-  # TODO: sync deleted contacts
+  # Deletes prospects from Pardot API that have been marked for deletion
+  # by the account deletion process.
+  def self.delete_pardot_prospects
+    pardot_ids_to_delete = ContactRollupsPardotMemory.where(delete_from_pardot: true).pluck(:pardot_id)
+    failed_deletion_pardot_ids = PardotV2.delete_prospects(pardot_ids_to_delete)
+
+    # The difference between these two arrays
+    # represents the IDs that were successfully deleted from Pardot.
+    pardot_ids_deleted = pardot_ids_to_delete - failed_deletion_pardot_ids
+
+    # clean-up step to delete rows once they've been deleted from Pardot
+    ContactRollupsPardotMemory.where(pardot_id: pardot_ids_deleted).delete_all
```

Comment:
Will there be a case that we delete a contact here, but the contact still exists in a source table somewhere (because we fail to find all references of that email in both dashboard and pegasus)? If so, the next time the pipeline runs, it will recreate the contact here and create a new Pardot prospect (with a new Pardot Id).